fastapi_etebase/routers/authentication.py

```diff
@@ -64,9 +92,10 @@ def send_user_logged_out_async(user: User, request: Request):
     user_logged_out.send(sender=user.__class__, request=request, user=user)
 
 
+@sync_to_async
 def validate_login_request(
-    validated_data: AuthenticationLoginInnerSerializer,
-    challenge_sent_to_user: AuthenticationLoginSerializer,
+    validated_data: Union[LoginResponse, ChangePasswordResponse],
+    challenge_sent_to_user: Union[Login, ChangePassword],
```

Comment:
Not a big deal, but: this doesn't really capture the correct typing, because you can have LoginResponse with Changepassword passed. It's good enough, so don't worry about it, but just mentioning it as I wonder if there's an easy way to encapsulate this information.

Reply:
Cleaner. :P

Reply:
check the last commit let me know if that solution works for now